gbpservice/neutron/services/servicechain/drivers/oneconvergence_intercloud_driver.py

```diff
@@ -241,12 +271,189 @@ def _fetch_template_and_params(self, context, sc_instance,
             for parameter in config_param_values.keys():
                 if parameter in node_params.keys():
                     stack_params[parameter] = config_param_values[parameter]
-        LOG.debug(stack_template)
+        LOG.debug(_("Final stack_template = %s, stack_params = %s"), stack_template, stack_params)
 
         return stack_template, stack_params
 
-    def _update_template_with_firewall_rules(self, context, provider_ptg,
-                                             provider_cidr, consumer_cidr,
+    def _get_hotplug_port_id(self, context, subnet_id):
+        """
+        First remove the router interface.
+        Then create a new port with the same ip address, and hotplug
+        it to the provided instance
+        """
+        hotplug_port = None
+        subnet = self._core_plugin.get_subnet(context._plugin_context, subnet_id)
+        network_id = subnet['network_id']
+        ports = self._core_plugin.get_ports(
+            context._plugin_context,
+            filters={'device_owner': ['network:router_interface']})
+        router_id = None
+        if ports:
+            for port in ports:
+               if port['fixed_ips'][0]['subnet_id'] == subnet_id:
+                   router_id = port['device_id']
+                   break
+        if router_id:
+	    with context._plugin_context.session.begin(subtransactions=True):
+                self._l3_plugin.remove_router_interface(context._plugin_context,
+                                                        router_id, {"subnet_id": subnet_id})
+
+        ip_address = subnet['gateway_ip']
+
+        port_name = "hotplug-" + ip_address
+        # mac_address can not be None
+        attrs = {'port': {'tenant_id': context._plugin_context.tenant_id,
+                 'name': port_name,
+                 'network_id': network_id,
+                 'fixed_ips': [{"ip_address": ip_address}],
+                 'port_security_enabled': False,
+                 'admin_state_up': True,
+                 'mac_address': attributes.ATTR_NOT_SPECIFIED,
+                 'device_id': '',
+                 'device_owner': ''
+                }}
+        with context._plugin_context.session.begin(subtransactions=True):
+          for i in xrange(5):
```

Comment:
Please add comment with FIXME about the issue here for  the retry with sleep so that we wont lose track of it